README.md

```diff
@@ -338,6 +339,34 @@ function stateReducer(state, changes) {
 }
 ```
 
+### keyDownHandlers
+
+> `{ ...handlers }` | optional
+
+Passing a `keyDownHandlers` object to this prop allows you to use your own key down handlers which can be handy if you want custom behavior.
+
+The current key down handlers that are defined in downshift are: `ArrowUp`, `ArrowDown`, `Enter`, and `Escape`. If a key and handler you pass is already defined in downshift, the custom handler you pass will be used instead.
+
+For example, you want a custom `Enter` handler:
+
+```jsx
+const handlers = {
+  Enter(event) {
+    // Add customized behavior for enter keypress below
+    if (this.isOpen) {
+      this.selectHighlightedItem({
+        type: Downshift.stateChangeTypes.keyDownEnter,
+      })
+    }
+  }
+}
```

Comment:
Does this example imply that developers must also include the existing handler logic if they write their own? Perhaps I'm misunderstanding something, but I thought that this solution allowed for the user to write their own handler without having to worry about what Downshift was doing with that handler previously.

For example, I would have thought I could just write

```js
const handlers = {
  Enter (event) {
    // my custom behavior for ENTER event
  }
}
```

And Downshift would have "merged" my custom handler with whatever it needed to do internally on its own.

Correct me if I'm wrong, but the way I'm reading this example it seems like the developer is having to do the "merging" themselves by including Downshift's existing logic at the bottom of the handler themselves (which I don't think would even work because `this` will be different depending on the context).

Reply:
No, that's not the intention of the feature. The intention is to allow you to completely override the behavior for a specific key. If you want downshift to merge behavior, then you'll use an `onKeyDown` event handler yourself. This is why I wanted to add [this note](https://github.com/paypal/downshift/pull/346#discussion_r170666325).

Reply:
Including Downshift's existing logic is optional (technically it is the existing logic minus the `e.preventDefault()`). From my understanding, `this` would never differ here - the object returned from `getStateAndHelpers()` is always bound.

Would updating the example in the doc from this:
```jsx
const handlers = {
  Enter(event) {
    // Add customized behavior for enter keypress below
    if (this.isOpen) {
      this.selectHighlightedItem({
        type: Downshift.stateChangeTypes.keyDownEnter,
      })
    }
  }
}

const ui = <Downshift keyDownHandlers={handlers} />
```

to this:
```jsx
const handlers = {
  Enter (event) {
    // my custom behavior for ENTER event
  }
}

const ui = <Downshift keyDownHandlers={handlers} />
```

be more clear?

Reply:
Maybe it would be worth adding to the docs something explicit such as...

"For example, if you create an Enter handler, this will not change the default behavior of the other keyDown events: ArrowUp, ArrowDown, and Escape."

Reply:
> From my understanding, this would never differ here - the object returned from getStateAndHelpers() is always bound.

Ah, my apologies. I see now you are explicitly binding the context on the handler invocation.